docs/managed-flags.rst

```diff
@@ -0,0 +1,86 @@
+.. _automatic-flags:
+
+Automatic Flags
+===============
+
+The reactive framework will automatically set some flags for your charm,
+based on lifecycle events from Juju.  These flags can inform your charm
+of things such as upgrades, config changes, relation activity, etc.
+
+With a few exceptions, noted below, these flags will be set by the framework
+but it is up to your charm to clear them, if need be.  To avoid conflicts
+between layers, it is recommended that only the top-level charm layer (or
+interface layer, in the case of endpoint flags) use any of the automatic flags
+directly; any base layer should instead use
+:func:`~charms.reactive.flags.register_trigger` to "wrap" the automatic flag
+with a layer-specific flag that can be safely used within that layer.
+
+The flags that are set by the framework are:
+
++----------------------------------------------+------------------------------------------------------------+
+| ``upgrade.series.in-progress``               | This is set when the ``pre-series-upgrade`` hook is        |
```

Comment:
Since these are new hooks, most charmers will probably first use them as a flag, not as a hook. Thus, the explanation of "this is set when x hook is fired" is mostly useless and requires the charmer to now look up this hook in the Juju docs in order to get more info. We should just copy the explanation of the hook in the Juju docs to here. Is there any draft for that explanation?

Also, I think it's best for the Juju docs to first explain the behavior of the flag (with code examples of a typical use case) and then mention the hook as a "low-level" way to do series migration for non-reactive, legacy charms.